src/Client/GetHedgedConnections.h

```diff
@@ -0,0 +1,173 @@
+#pragma once
+
+#if defined(OS_LINUX)
+
+#include <Common/TimerDescriptor.h>
+#include <Client/ConnectionPoolWithFailover.h>
+#include <Core/Settings.h>
+#include <Common/Epoll.h>
+#include <unordered_map>
+#include <memory>
+
+namespace DB
+{
+
+using TimerDescriptorPtr = std::shared_ptr<TimerDescriptor>;
+
+/// Class for establishing hedged connections with replicas.
+/// It works with multiple replicas simultaneously without blocking by using epoll.
+class GetHedgedConnections
```

Comment:
It is better to use known for class name. E.g. it may be `HedgedConnectionsGetter` or `HedgedConnectionsFactory` or probably `HedgedConnectionsManager`